src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -283,21 +304,19 @@ private UnboundLambda BindAnonymousFunction(AnonymousFunctionExpressionSyntax sy
             var data = lambda.Data;
             if (data.HasExplicitlyTypedParameterList)
             {
+                int firstDefault = -1;
                 for (int i = 0; i < lambda.ParameterCount; i++)
                 {
-                    // UNDONE: Where do we report improper use of pointer types?
-                    var type = data.ParameterTypeWithAnnotations(i).Type;
-                    if (type is { })
+                    // paramSyntax should not be null here; we should always be operating on an anonymous function which will have parameter information
+                    var paramSyntax = lambda.ParamSyntax(i);
+                    if (paramSyntax!.Default != null && firstDefault == -1)
                     {
-                        if (type.IsStatic)
-                        {
-                            Error(diagnostics, ErrorFacts.GetStaticClassParameterCode(useWarning: false), syntax, type);
-                        }
-                        if (data.Scope(i) == DeclarationScope.ValueScoped && !type.IsErrorTypeOrRefLikeType())
-                        {
-                            diagnostics.Add(ErrorCode.ERR_ScopedRefAndRefStructOnly, data.ParameterLocation(i));
-                        }
+                        firstDefault = i;
                     }
+
+                    // UNDONE: Where do we report improper use of pointer types?
```

Comment:
>UNDONE [](http://example.com/codeflow?start=23&length=6)

Consider using `PROTOTYPE` in the feature branch instead, since we automatically catch `PROTOTYPE` comments before merging into main.

Reply:
This is copied from the original implementation, it's not newly introduced in this branch.

Reply:
I'm happy to change this though if a different flag would be better

Reply:
No, it's not a new thing, so let's leave it as is. We have no plans to address it.